filters/openpolicyagent/preprocessor.go

```diff
@@ -41,9 +44,18 @@ func (p *opaPreProcessor) Do(routes []*eskip.Route) []*eskip.Route {
 		p.preloadInstancesParallel(bundleConfigs)
 	})
 
+	// check already processed bundles
+	bundles := make([]string, 0, len(bundleConfigs))
+	for _, bundleName := range bundleConfigs {
+		if _, ok := p.bundleMap[bundleName]; !ok {
+			bundles = append(bundles, bundleName)
+			p.bundleMap[bundleName] = struct{}{}
```

Comment:
This looked good me until I check the test failure.
We currently have a retry possibility for tasks which failed to get added to the background queue when it is full(those are retried with the next route update), which is lost with this. Except for this I think the behavior is synonyms for sync.Once. 
Given these impacts I think sync.Once is simpler approach and the original issue I reported was misleading.